### PR TITLE
Update project for pre-commit linting and testing

### PR DESCRIPTION
**Pull request by likwid**
* Update image adapter to use type instead of name for accuracy
* Made some changes to the specs to pass linter
* Add test-watch command for development
* Remove watch flag from test for pre-commit hook
* Add implicit return for promises in tests

### CODE REVIEW

**Comment by likwid** [comment]
@theqabalist testing PR notiication

File: package.json [hunk 1]
```diff
@@ -5,16 +5,20 @@
...
   "description": "A simple helpful robot for your Company",
   "scripts": {
+    "coverage": "istanbul cover --include-all-sources --root lib _mocha -- --reporter nyan 'spec/**/*.spec.js'",
     "hubot": "hubot --adapter slack",
-    "test": "mocha --watch spec/**/*.spec.js --reporter min",
-    "coverage": "istanbul cover --include-all-sources --root lib _mocha -- --reporter nyan 'spec/**/*.spec.js'"
+    "lint": "eslint .",
+    "pre-commit-msg": "echo 'Pre-commit linting and unit testing' && exit 0",
+    "test": "mocha spec/**/*.spec.js --reporter min",
+    "test-watch": "mocha --watch spec/**/*.spec.js --reporter min"
```
Comment (theqabalist) [inline package.json]: This is just a small thing, but I'm not sure that I would make the automated one the nice one (test has special semantics [npm test vs npm run test-watch]).  You can make the special named one like pre-commit-test, and use like the dot reporter for it.

Comment (theqabalist) [reply package.json]: Also I haven't been on github in so long, the PR stuff is a lot different now.

**Review by theqabalist — COMMENTED** [review]
Summary.

**Comment by theqabalist** [comment]
Guess they are trying to get more formal in the PR stuff, like bitbucket or whatever.

File: package.json (hunk 1 above)
Comment (likwid) [reply package.json]: I don't know what the dot reporter is. You set all that up :D, I don't really use it. I am amenable to changing it for the pre-commit stuff.

**Comment by theqabalist** [comment]
It's just reporter dot instead of reporter min.

**Pull request merged**